#982 Fix search bar

Priyal208 opened the pull request:
## Description

As mentioned in issue #982  assigned to me under gssoc'24 , I optimised the search bar which earlier gave not so appropriate results making it more user friendly.

## Related Issues

#982 

## Changes Proposed

Additionaly I added a drop which allows users to select whether they want to serach by name, location or skill and then it gives specific results as per requirement only unlike earlier random results. 

## Checklist

- [x] I have read and followed the [Contribution Guidelines](https://github.com/shyamtawli/devFind/blob/master/CONTRIBUTING.md).
- [x] All new and existing tests passed.
- [x] I have updated the documentation to reflect the changes I've made.
- [x] My code follows the code style of this project.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

![Screenshot 2024-07-10 191035](https://github.com/shyamtawli/devFind/assets/135015851/284d5d61-a812-417a-b809-7aee939b0bdc)
![Screenshot 2024-07-10 191050](https://github.com/shyamtawli/devFind/assets/135015851/4fd388f6-1f8f-4dbe-86c6-49309b5a0047)
![Screenshot 2024-07-10 191107](https://github.com/shyamtawli/devFind/assets/135015851/14cdce35-aac4-49a4-95d9-3828c07b036f)

## Note to reviewers

Kindly review it asap and give me appropriate level as I also additionaly added dropdown bar and made it most optimized. Thank you

Priyal208 commented:
@shyamtawli kindly review it

akbatra567 commented:
@Priyal208, Could you look into this scenario ?

We have a search string 
1. `kolkata,india`
2. `kolkata, india`
Just a difference of space is a giving us different result.

![Screenshot 2024-07-13 at 12 50 32 AM](https://github.com/user-attachments/assets/710a75ad-b86d-4bda-9c10-ba275998a4cc)
![Screenshot 2024-07-13 at 12 50 15 AM](https://github.com/user-attachments/assets/54d095f0-874a-4b90-b963-6f0456e52146)

Issues with this is even if we have a match for the location, we won't be able to find if a comma or space is in different order. 
P.S: This is one scenario for location, do check for other filters as well.

Priyal208 commented:
@shyamtawli now pls you can check i fixed that problem of space giving different result for all test cases
You can see search queries before space and after space
![Screenshot 2024-07-15 185710](https://github.com/user-attachments/assets/648e1f94-3936-4d20-a9d3-6ea76461cc21)
![Screenshot 2024-07-15 185719](https://github.com/user-attachments/assets/32c98d3d-07b5-4f25-b251-8d737265d945)

akbatra567 commented:
Hey @shyamtawli, 
There is a scenario which doesn't work as of now:  
* If we would like to search multiple skills at once.
* For example: Javascript, React

Do you need this to be handle in this PR, or we can create a new issue for that.

Priyal208 commented:
@shyamtawli can you pls review this

shyamtawli reviewed (changes requested):
@Priyal208 Can you improve the UI, currently it is not looking decent.

Priyal208 commented:
@shyamtawli i did improve the UI u can see i added appropriate spacing, kindly merge the PR now, also pls see that you are assigning appropriate level as I did improve search operation, also added dropdown, even improved the UI
![Screenshot 2024-07-22 125314](https://github.com/user-attachments/assets/ce0a5e99-e2b9-4794-9fdb-7f3f496fc60a)
![Screenshot 2024-07-22 125322](https://github.com/user-attachments/assets/8fb83b11-0aec-4896-bdd3-2c573c65ee72)

Priyal208 commented:
@shyamtawli sure thanks
